File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode : public rclcpp::Node 
{
    private:
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher;
        rclcpp::TimerBase::SharedPtr timer;
        std::string robot_name;

        void callbackSubscribed() 
        {
            auto msg = example_interfaces::msg::String();
            msg.data = std::string("Hi, this is ") + this->robot_name + std::string(" from the robot news station");
            this->publisher->publish(msg);
        }

    public:
        RobotNewsStationNode() : Node("robot_news_station")
        {
            // Declare runtime parameters with default value
            this->declare_parameter("robot_name", "R2D2");

            // Get Runtime Parameters
            this->robot_name = this->get_parameter("robot_name").as_string();

            this->publisher = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
            this->timer = this->create_wall_timer(
                                    std::chrono::milliseconds(500),
                                    std::bind(&RobotNewsStationNode::callbackSubscribed, this)
                                );

            RCLCPP_INFO(this->get_logger(), "Robot news station is broadcasting...");
        }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3

# Import ROS2 libraries
import rclpy
from rclpy.node import Node

# Import messages
from example_interfaces.msg import String

class RobotNewsStationNode(Node):

    def __init__(self):
        super().__init__('robot_news_station')

        self.__robot_name = "C3P0"
        self.__publisher = self.create_publisher(String, 'robot_news', 10)
        self.__logger = self.get_logger()
        self.__timer = self.create_timer(0.5, self.publish_news)

        self.__logger.info("Robot News Station has been started")

    
    def publish_news(self):
        msg = String()
        msg.data = f"Hi, this is {self.__robot_name}, from the robot news station"

        # publish the msg
        self.__publisher.publish(msg)

        # log the msg on the console
        # self.get_logger().info(msg.data)


def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()